types/sinon/ts3.1/sinon-tests.ts

```diff
@@ -427,7 +426,7 @@ function testStub() {
     const obj = class {
         foo() { }
         promiseFunc() { return Promise.resolve('foo'); }
-        promiseLikeFunc() { return {} as any as Bluebird<string>; }
+        promiseLikeFunc() { return Promise.resolve('foo') as PromiseLike<string>; }
```

Comment:
i would suggest putting it in a const to make this very clear and avoiding casting it.

```ts
const p: PromiseLike<string> = Promise.resolve('foo');
return p;
```

Reply:
Could you explain the difference between casting and declaration with explicit type in this case?

As far as I understand, type checks are the same, e.g. one cannot cast an object that does not implement the interface.

Reply:
There's likely very little difference, if any (I would have to do some investigation to be sure). it was more to make it clearer, but now that I read it again, it seems fine actually as we'd just be introducing bloat.